sdkv2/helper.go

```diff
@@ -433,3 +464,119 @@ func (lb *Helper) queryPlanAPIOption() func(*middleware.Stack) error {
 		return nil
 	}
 }
+
+// SetPartitionKeyInfo stores partition key information for a table in a thread-safe manner.
+func (lb *Helper) SetPartitionKeyInfo(tableName string, keys []string) {
+	lb.pkInfoMutex.Lock()
+	defer lb.pkInfoMutex.Unlock()
+	lb.pkInfoPerTable[tableName] = keys
+}
+
+// GetPartitionKeyInfo retrieves partition key information for a table in a thread-safe manner.
+func (lb *Helper) GetPartitionKeyInfo(tableName string) ([]string, bool) {
+	lb.pkInfoMutex.RLock()
+	defer lb.pkInfoMutex.RUnlock()
+	keys, ok := lb.pkInfoPerTable[tableName]
+	return keys, ok
+}
+
+func (lb *Helper) extractPartitionKey(
+	item map[string]types.AttributeValue,
+	tableName string,
+) map[string]types.AttributeValue {
+	pkInfo, exists := lb.GetPartitionKeyInfo(tableName)
+	if !exists {
+		return nil
+	}
+
+	partitionKey := make(map[string]types.AttributeValue)
+	for _, keyName := range pkInfo {
+		if val, ok := item[keyName]; ok {
+			partitionKey[keyName] = val
+		}
+	}
+
+	if len(partitionKey) == 0 {
+		return nil
+	}
+	return partitionKey
+}
+
+func (lb *Helper) hashPartitionKey(partitionKey map[string]types.AttributeValue, tableName string) int64 {
```

Comment:
This guys should return error if it can't get a hash, 0 hash is also a valid hash value.
Also can you make it use a hash writer as I mentioned earlier, make mumur3 implement `hash.Hash64` and dump all the data into it, this way we will save memory from being allocated multiple times.